yellowbrick/features/pcoords.py

```diff
@@ -268,32 +295,28 @@ def draw(self, X, y, **kwargs):
         )
         colors = dict(zip(self.classes_, color_values))
 
-        # Track which labels are already in the legend
-        used_legends = set([])
-
         # TODO: Make this function compatible with DataFrames!
         # TODO: Make an independent function to allow addition of instances!
-        for idx, row in enumerate(X):
-            # TODO: How to map classmap to labels?
-            label = y[idx] # Get the label for the row
-            label = self.classes_[label]
-
-            if label not in used_legends:
-                used_legends.add(label)
-                self.ax.plot(x, row, color=colors[label], alpha=0.25, label=label, **kwargs)
-            else:
-                self.ax.plot(x, row, color=colors[label], alpha=0.25, **kwargs)
+        x_separated = hstack([X, ones((nrows, 1))])
+        increments_separated = increments.copy()
+        increments_separated.append(None)
+        for label, color in colors.items():
+            x_in_class = x_separated[y == label, :]
+            increments_in_class = increments_separated * len(x_in_class)
+            if len(x_in_class) > 0:
+                self.ax.plot(increments_in_class, x_in_class.flatten(), label=label, color=colors[label],
+                             alpha=0.5, linewidth=1, **kwargs)
```

Comment:
Would you mind walking me through this new drawing method? I can see that the overall goal here was to call the plot method once per class instead of once per row - a laudable goal and one that I can see dramatically improving the performance of the visualizer. It might help to have a few comments to guide other contributors to this method, and if I could get a more detailed sense of what you're doing I can work to apply it to other visualizers. 

The alpha is set here to 0.5 -- so I'm not sure I understand exactly the change in the opacity style. Given the two figures you presented, I believe the top figure is preferable because it shows darker regions of greater instance density on a per-class basis. Was the top figure generated with this code or the old code? 

Frankly, even if the bottom figure was generated with your revisions, I think these revisions are worth it to keep the opacity changes and we can work on detailed styling in another PR.

Reply:
Sounds good.  I will add comments to the code and some documentation to this PR to clarify.

The top image was the "current" method.  The bottom image was the "new" method.

I'll revert back to 0.25 alpha.  The 0.5 seemed to give a better visual agreement with the former style.  Are you OK with the linewidth specification?  Again, this gives better stylistic agreement for the samples I investigated.

Reply:
I'm ok with the `linewidth` argument -- in the past, we've had issues with fixed settings like this (e.g. in a "poster" mode of matplotlib where rcParams are set, this fixed value might cause some issues) but I think it will be ok for now.